### PR TITLE
FALCON-1858: Support HBase as a storage backend for Falcon Titan graphDB

### PR DESCRIPTION
**Pull request by vrangan**
This patch provides the following features
1.   Ability to build Falcon with storage backend plugins without the artifacts (hbase and bdb).
2.   Ability to package a standalone HBase if desired
3.  Backport of Titan 1.0 Hbase 1.0 storage backend into Titan 0.5.4 (shaded classes - mostly similar to Apache Atlas usage of the same) (most of the addons/titan codebase)
4.  Default build profile is with bdb backend with bdb artifacts also included in the build.

Tested the HBase backend with 1.1.5 and 1.2.1 Hbase environments.   Ran test-patch successfully with enabling/disabling of bdb db artifacts and also with all potential addons enabled

### CODE REVIEW

File: src/conf/startup.properties [hunk 1]
```diff
@@ -168,9 +168,20 @@ prism.configstore.listeners=org.apache.falcon.entity.v0.EntityGraph,\
...
 # Graph Storage
-*.falcon.graph.storage.directory=/${falcon.home}/data/graphdb
-*.falcon.graph.storage.backend=berkeleyje
+# Berkeley DB
+#*.falcon.graph.storage.directory=/${falcon.home}/data/graphdb
+#*.falcon.graph.storage.backend=berkeleyje
+
+
+# Enabled the following for HBase
+*.falcon.graph.storage.backend=hbase
```
Comment (peeyushb) [inline src/conf/startup.properties]: Here hbase as storage backend is enabled. I am of the opinion that Berkley should be enabled by default.

**Comment by peeyushb** [comment]
@vrangan  I have done review of the pull request. On revieweing I found that docs, how to build the Falcon with hbase or with berkley profile to use it as a storage backend, is not available.

File: src/conf/startup.properties (hunk 1 above)
Comment (vrangan) [inline src/conf/startup.properties]: We should not have either of them enabled by default in the release versions and have users download the necessary jars.  Have updated the docs

**Comment by peeyushb** [comment]
When I have used the patch from this pull request , Falcon build is proper. But when I am trying to start the Falcon server it is not starting on standalone machine.

**Comment by vrangan** [comment]
Discussed with @peeyushb  - we have one hbase property enabled by default in startup.properties.   That requires standalone hbase with default properties (as the other hbase related properties are commented out).  So, as per comment in the properties we will comment all the graphdb related properties and users will have to uncomment one or the other backend.

Note that hadoop installer can make this automatic without users needing to do it.

**Comment by vrangan** [comment]
Made a change after discussion with @peeyushb.   Will follow up with a documentation JIRA to augment how to standup a hbase standalone mode cluster

**Comment by peeyushb** [comment]
Patch looks good. +1

**Comment by bvellanki** [comment]
+1

**Comment by peeyushb** [comment]
I will commit this patch soon, if no further comments.

**Pull request closed without merging**